PR: https://github.com/odoo/odoo/pull/185243

From: c96acd45acf05a8ea798ae95b1fd5994e7e2e546
From: jugj-odoo
Date: 2025-06-17 19:08:50

Breaking data model changes scores: del:4 + add:0, change matches:
-    in_country_name = fields.Char(string="Country", help="Based on IP Address", readonly=True)
-    in_city = fields.Char(string="City", readonly=True)
-    out_country_name = fields.Char(help="Based on IP Address", readonly=True)
-    out_city = fields.Char(readonly=True)

Total Changes: 14

[IMP] hr_attendance,base_geolocalize: Add georeverse lookup

We allow in this commit to get the precise location of an attendance if GPS coordinates are
available. We also add the possibility of general georeverse lookup from latlong coordinates.
task-4284048

closes odoo/odoo#185243

Related: odoo/upgrade#6657
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_attendance/models/hr_attendance.py
+++ b/addons/hr_attendance/models/hr_attendance.py
@@ -54,8 +54,7 @@ class HrAttendance(models.Model):
     no_validated_overtime_hours = fields.Boolean(compute='_compute_no_validated_overtime_hours')
     in_latitude = fields.Float(string="Latitude", digits=(10, 7), readonly=True, aggregator=None)
     in_longitude = fields.Float(string="Longitude", digits=(10, 7), readonly=True, aggregator=None)
-    in_country_name = fields.Char(string="Country", help="Based on IP Address", readonly=True)
-    in_city = fields.Char(string="City", readonly=True)
+    in_location = fields.Char(help="Based on GPS-Coordinates if available or on IP Address")
     in_ip_address = fields.Char(string="IP Address", readonly=True)
     in_browser = fields.Char(string="Browser", readonly=True)
     in_mode = fields.Selection(string="Mode",
@@ -67,8 +66,7 @@ class HrAttendance(models.Model):
                                default='manual')
     out_latitude = fields.Float(digits=(10, 7), readonly=True, aggregator=None)
     out_longitude = fields.Float(digits=(10, 7), readonly=True, aggregator=None)
-    out_country_name = fields.Char(help="Based on IP Address", readonly=True)
-    out_city = fields.Char(readonly=True)
+    out_location = fields.Char(help="Based on GPS-Coordinates if available or on IP Address")
     out_ip_address = fields.Char(readonly=True)
     out_browser = fields.Char(readonly=True)
     out_mode = fields.Selection(selection=[('kiosk', "Kiosk"),
@@ -600,8 +598,8 @@ class HrAttendance(models.Model):
                 'out_longitude': city_data['longitude'],
                 'in_latitude': city_data['latitude'],
                 'out_latitude': city_data['latitude'],
-                'in_city': city_data['city'],
-                'out_city': city_data['city'],
+                'in_location': city_data['city'],
+                'out_location': city_data['city'],
                 'in_ip_address': "127.0.0.1",
                 'out_ip_address': "127.0.0.1",
                 'in_browser': 'chrome',
@@ -616,8 +614,8 @@ class HrAttendance(models.Model):
                 'out_longitude': city_data['longitude'],
                 'in_latitude': city_data['latitude'],
                 'out_latitude': city_data['latitude'],
-                'in_city': city_data['city'],
-                'out_city': city_data['city'],
+                'in_location': city_data['city'],
+                'out_location': city_data['city'],
                 'in_ip_address': "127.0.0.1",
                 'out_ip_address': "127.0.0.1",
                 'in_browser': 'chrome',
